packages/components/app/styles/components/flyout.scss

```
@@ -0,0 +1,96 @@
+//
+// FLYOUT
+//
+
+.hds-flyout {
+  z-index: 50;
+  flex-direction: column;
+  padding: 0;
+  background: var(--token-color-surface-primary);
+  border: none;
+  box-shadow: 0 2px 3px 0 #3b3d4540, 0 12px 24px 0 #3b3d4559; // TODO: convert to token
+
+  &[open] {
+    position: fixed;
+    display: flex;
+    height: 100vh;
+    max-height: 100vh;
+    margin: 0;
+  }
+
+  // we hide the native `::backdrop` pseudo-element in favor of using
+  // an `hds-flyout__overlay` element to detect click events with more ease
+  &::backdrop {
+    display: none;
+  }
+}
+
+.hds-flyout__overlay {
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  z-index: 50;
+  background: var(--token-color-palette-neutral-700);
+  opacity: 0.5;
+}
+
+.hds-flyout__header {
+  display: flex;
+  flex: none;
+  align-items: flex-start;
+  padding: 16px 24px;
+  color: var(--token-color-foreground-strong);
+}
+
+.hds-flyout__icon {
+  flex: none;
+  align-self: center;
+  margin-right: 16px;
+}
+
+.hds-flyout__title {
+  flex-grow: 1;
+}
+
+.hds-flyout__tagline {
+  margin-bottom: 4px;
+  color: var(--token-color-foreground-faint);
+}
+
+.hds-flyout__dismiss {
+  align-self: center;
+  margin-left: 16px;
+}
+
+.hds-flyout__description {
+  padding: 0 24px 16px;
+  color: var(--token-color-foreground-primary);
+}
+
+.hds-flyout__body {
+  flex: 1 1 auto;
+  padding: 24px;
+  overflow-y: auto;
+  overscroll-behavior: contain;
+  border-top: 1px solid var(--token-color-border-primary);
+}
+
+.hds-flyout--size-medium {
+  width: min(480px, calc(100vw - 40px));
+  max-width: calc(100vw - 40px);
+
+  &[open] {
+    margin-left: calc(100% - min(480px, calc(100vw - 40px)));
+  }
+}
+
+.hds-flyout--size-large {
+  width: min(720px, calc(100vw - 40px));
+  max-width: calc(100vw - 40px);
+
+  &[open] {
+    margin-left: calc(100% - min(720px, calc(100vw - 40px)));
+  }
+}
```

What do you think, would this make it easier to maintain/read/understand? (I have mixed feelings)
```suggestion
.hds-flyout {
  --max-width-responsive: calc(100vw - 40px);
  --width-calculated: min(var(--min-width-fixed), var(--max-width-responsive));
  width: var(--width-calculated);
  max-width: var(--max-width-responsive);

  &[open] {
    margin-left: calc(100% - var(--width-calculated));
  }
}

.hds-flyout--size-medium {
  --min-width-fixed: 480px;
}

.hds-flyout--size-large {
  --min-width-fixed: 720px;
}
```

Notice: variable names are tentative, and where the declarations leave in the CSS depends (may be unified in the top `.hds-flyout` block?

---

Used a mixin here to avoid code duplication and keep the calculations in sync.

---

Neat!